packages/valory/skills/decision_maker_abci/behaviours/storage_manager.py

```diff
@@ -253,19 +270,102 @@ def _fetch_accuracy_info(self) -> Generator[None, None, bool]:
+        if tool_accuracy_data:
+            self.accuracy_information = tool_accuracy_data
+
+    def _read_local_policy_date(self) -> int:
+        """Reads the updated timestamp from the policy file."""
+
+        self.context.logger.info("Reading policy updated timestamp...")
+        policy = self._read_tool_accuracy_local()
+        if policy:
+            policy_json = json.loads(policy)
+            updated_timestamp = policy_json.get("updated_ts")
+
+            if updated_timestamp is str:
+                self.context.logger.info(
+                    f"Local updated timestamp found {updated_timestamp}"
+                )
+                return updated_timestamp
+            else:
+                self.context.logger.info("No timestamp found. Using minium: 0")
+                return 0
+        return 0
+
+    def _fetch_remote_tool_date(self) -> Generator[None, None, int]:
+        """Fetch the max transaction date from the remote accuracy storage."""
+        self.context.logger.info("Checking remote accuracy information date... ")
+        self.context.logger.info("Trying to read max date in file...")
+        tool_accuracy_data = yield from self._fetch_accuracy_info()
+
+        if tool_accuracy_data:
+            sep = self.acc_info_fields.sep
+            tool_accuracy_data.seek(0)  # Ensure we’re at the beginning
+            reader = csv.DictReader(tool_accuracy_data.readlines(), delimiter=sep)
+
+        max_transaction_date = None
+
+        # try to read the maximum transaction date in the remote accuracy info
+        try:
+            for row in reader:
+                current_transaction_date = row.get("max")
+                if (
+                    max_transaction_date is None
+                    or current_transaction_date > max_transaction_date
+                ):
+                    max_transaction_date = current_transaction_date
+
+        except TypeError:
+            self.context.logger(
+                "Invalid transaction date found. Continuing with local accuracy information..."
+            )
+            return 0
+
+        if max_transaction_date:
+            self.context.logger.info(f"Maximum date found: {max_transaction_date}")
+            format_str = "%Y-%m-%d %H:%M:%S"
+            max_datetime = datetime.strptime(max_transaction_date, format_str)
+            unix_timestamp = int(max_datetime.timestamp())
+            print(f"returning timestamp: {unix_timestamp}")
+            return unix_timestamp
+
+        else:
+            self.context.logger.info("No maximum date found.")
+            return 0
+
+    def _check_local_policy_store_overwrite(self) -> Generator[None, None, bool]:
+        """Compare the local and remote policy store dates and decide which to use."""
+
+        local_policy_store_date = self._read_local_policy_date()
+        remote_policy_store_date = yield from self._fetch_remote_tool_date()
+
+        three_days_in_seconds = 3 * 24 * 3600
```

Comment:
Please make this configurable.

Reply:
Please explain what you mean.

Reply:
You can add this duration to the agent's configuration so that it can be controlled without code changes.